File: Java/src/pattens/strategy/SimUDuck.java
package pattens.strategy;

public class SimUDuck {
	public static void main(String argv[]) {
		MallardDuck mallard = new MallardDuck();
		mallard.display();
		mallard.quack();
		mallard.swim();
		mallard.fly();
		
		RedheadDuck redhead = new RedheadDuck();
		redhead.display();
		redhead.quack();
		redhead.swim();
		redhead.fly();
		
		RubberDuck rubber = new RubberDuck();
		rubber.display();
		rubber.quack();
		rubber.swim();
		rubber.fly();
		
		DecoyDuck decoy = new DecoyDuck();
		decoy.display();
		decoy.quack();
		decoy.swim();
		decoy.fly();
	}
}

File: cpp/src/patterns/strategy/SimUDuck.cc
#include "Duck.h"

int main () {
    RedheadDuck *redhead = new  RedheadDuck();
    redhead->display();
    redhead->swim();
    redhead->fly();
    redhead->sound();

    MallardDuck *mallard = new MallardDuck();
    mallard->display();
    mallard->swim();
    mallard->fly();
    mallard->sound();

    RubberDuck *rubber = new RubberDuck();
    rubber->display();
    rubber->swim();
    rubber->fly();
    rubber->sound();

    DeconyDuck *decony = new DeconyDuck();
    decony->display();
    decony->swim();
    decony->fly();
    decony->sound();
    return 0;
}
